crates/client/src/client/schema/tx.rs

```diff
@@ -187,8 +190,11 @@ impl TryFrom<ProgramState> for fuel_vm::ProgramState {
 pub enum TransactionStatus {
     SubmittedStatus(SubmittedStatus),
     SuccessStatus(SuccessStatus),
+    SuccessDuringBlockProductionStatus(SuccessDuringBlockProductionStatus),
```

Comment:
Should we use a name which relates more closely to the pre-confirmations? For example `SuccessPreConfirmed`, `FailurePreConfirmed`, etc.

cc: @xgreenx

Reply:
Renamed in https://github.com/FuelLabs/fuel-core/pull/2752/commits/a473dbb947220b4961ce634ee400d6ecbefe5d44, as agreed with SDK.

The only minor drawback I see is that in case in the future other components than BlockProducer are able to create preconfirmations we'll not carry the information about the preconfirmation source. Most likely a YAGNI case, that's why I didn't add any kind of "source" field to preconfirmations.